src/test/java/de/wejago/hichi2influxDB/unit/MqttSubscriberServiceTest.java

```diff
@@ -0,0 +1,167 @@
+import java.time.Instant;
+import org.eclipse.paho.client.mqttv3.IMqttClient;
+import org.eclipse.paho.client.mqttv3.IMqttMessageListener;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.internal.util.reflection.Whitebox;
+import org.springframework.boot.test.system.CapturedOutput;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+
+@ExtendWith(OutputCaptureExtension.class)
+public class MqttSubscriberServiceTest {
+    @Mock
+    private InfluxDbService influxDbService;
+
+    @Mock
+    private IMqttClient mqttClient;
+    @InjectMocks
+    private MqttSubscriberService mqttSubscriberService;
+
+    private SensorEntry testSensorEntry;
+
+    private MqttMessage mqttMessage;
+
+    private Point point;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        point = generateTestPoint();
+    }
+
+    @Test
+    void testConnectSuccessfully(CapturedOutput output) throws MqttException {
+        // GIVEN
+        Whitebox.setInternalState(mqttSubscriberService, "brokerIp", "localhost");
+        Whitebox.setInternalState(mqttSubscriberService, "brokerPort", "1883");
+        Whitebox.setInternalState(mqttSubscriberService, "username", "user123");
+        Whitebox.setInternalState(mqttSubscriberService, "password", "test-pass");
+
+        // WHEN
+        mqttSubscriberService.connect();
+
+        // THEN
+        assertThat(output.getOut()).contains("Successfully connected to MQTT client!");
+    }
+
+    @Test
+    void testConnectFailureOnWrongPort(CapturedOutput output) {
+        // GIVEN
+        MqttSubscriberService mqttSubscriberService = new MqttSubscriberService();
+        Whitebox.setInternalState(mqttSubscriberService, "brokerIp", "localhost");
+        Whitebox.setInternalState(mqttSubscriberService, "brokerPort", "8080");
+        Whitebox.setInternalState(mqttSubscriberService, "username", "user123");
+        Whitebox.setInternalState(mqttSubscriberService, "password", "test-pass");
+
+        // WHEN
+        assertDoesNotThrow(() -> mqttSubscriberService.connect(), "The connect() method should not throw any exceptions");
+
+        // THEN
+        assertThat(output.getOut()).contains("Error connecting to MQTT client!");
+    }
+
+    @Test
+    public void testSubscribeWhenReceivingSkippedMessage() throws Exception {
+        //GIVEN
+        String testTopic = "test/topic";
+        testSensorEntry = this.generateTestSensorEntry();
+        mqttMessage = this.generateSkippedMqttMessage();
+        when(influxDbService.writePoint(testSensorEntry)).thenReturn(point);
```

Comment:
Till line 101 it's the same code as in the next test, extract a method.

Reply:
outdated